src/lib/fauna-client.js

```diff
@@ -0,0 +1,71 @@
+const http2 = require("http2");
+
+// Copied from the fauna-js driver:
+// https://github.com/fauna/fauna-js/blob/main/src/http-client/node-http2-client.ts
+
+module.exports = class FaunaClient {
+  constructor(endpoint, secret) {
+    this.session = http2
+      .connect(endpoint, {
+        peerMaxConcurrentStreams: 50,
+      })
+      .once("error", () => this.close())
+      .once("goaway", () => this.close());
+    this.secret = secret;
+  }
+
+  async query(query, format = "simple", typecheck = undefined) {
+    return new Promise((resolvePromise, rejectPromise) => {
+      let req;
+      const onResponse = (http2ResponseHeaders) => {
+        const status =
+          http2ResponseHeaders[http2.constants.HTTP2_HEADER_STATUS];
+        let responseData = "";
+
+        req.on("data", (chunk) => {
+          responseData += chunk;
+        });
+
+        req.on("end", () => {
+          resolvePromise({
+            status,
+            body: JSON.parse(responseData),
+            headers: http2ResponseHeaders,
+          });
+        });
+      };
+
+      try {
+        const httpRequestHeaders = {
+          Authorization: `Bearer ${this.secret}`,
+          "x-format": format,
+          "X-Fauna-Source": "Fauna Shell",
+          [http2.constants.HTTP2_HEADER_PATH]: "/query/1",
+          [http2.constants.HTTP2_HEADER_METHOD]: "POST",
+          ...(typecheck && { "x-typecheck": typecheck }),
+        };
+
+        req = this.session
+          .request(httpRequestHeaders)
+          .setEncoding("utf8")
+          .on("error", (error) => rejectPromise(error))
+          .on("response", onResponse);
+
+        req.write(JSON.stringify({ query }), "utf8");
+
+        // req.setTimeout must be called before req.end()
+        req.setTimeout(5000, () => {
```

Comment:
We should set the client timeout to the `--timeout` parameter + 5000 ms. If no `--timeout` is provided, then maybe 10000 ms is okay.

According to docs, if no `--timeout` is provided, then we should let the query run. Default query timeout for v10 is 5 seconds, so client timeout of 5 seconds in that case is too short.

https://docs.fauna.com/fauna/current/build/tools/shell/commands/eval#options